README.md

```diff
@@ -180,6 +180,51 @@ export https_proxy=<user>:<pass>@<proxy IP>:<proxy port>
 export https_proxy=<domain>%5C<user>:<pass>@<proxy IP>:<proxy port>
 ```
 
+#### Authenticate using Service Principal Names
+
+**1**. Create a new app registration [in the azure portal](https://ms.portal.azure.com/#blade/Microsoft_AAD_IAM/ApplicationsListBlade).
+
+**2**. Ensure the app registration is labeled as a Web App/API. The sign-on URL does not matter, simply set it to http://whatever
+
+![Example registration](images/registration.png)
+
+**3**. Add public keys or client secrets under the "keys" section of the application.
+
+![Keys](images/keys.png)
+
+To generate a private & public keypair for AzCopy, perform the following steps in a command line, replacing example-app with your app name:
+
```

Comment:
Would it be simpler to refer to [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest#certificate-based-authentication) instead?

Reply:
Azure CLI is now referred to within the readme, while maintaining the alternative setup.